packages/server/worker/src/lib/executors/webhook-job-executor.ts

```diff
@@ -141,14 +142,14 @@ async function handleSampleData(
     workerToken: string,
     log: FastifyBaseLogger,
 ): Promise<void> {
-    const { flowId, payload } = data
+    const { flowId, payload, flowVersionIdToRun } = data
     const latestFlowVersion = await engineApiService(
         engineToken,
         log,
     ).getFlowWithExactPieces({
         flowId,
         type: GetFlowVersionForWorkerRequestType.LATEST,
```

Comment:
I don't think you need the type GetFlowVersionForWorkerRequestType to fetch the flow, I would edit the parent endpoint since you already have flowId, and versionId you can fetch it directly in the endpoint.